plugin/hover.py

```diff
@@ -215,6 +216,33 @@ def diagnostics_content(self) -> str:
             formatted.append("</div>")
         return "".join(formatted)
 
+    def document_link_content(self, listener: AbstractViewListener, point: int) -> str:
+        if userprefs().link_highlight_style not in ("underline", "none"):
+            return ""
+        contents = []
+        for sv in listener.session_views_async():
+            if sv.has_capability_async("documentLinkProvider"):
+                link = sv.session_buffer.get_document_link_at_point(sv.view, point)
+                if link is not None:
+                    target = link.get("target")
+                    if not target:
+                        if not sv.has_capability_async("documentLinkProvider.resolveProvider"):
+                            continue
+                        sv.session.send_request_task(Request.resolveDocumentLink(link, sv.view)).then(
+                            lambda result: self._on_resolved_link(sv.session_buffer, result))
+                        link = sv.session_buffer.get_document_link_at_point(sv.view, point)  # resolved link ???
```

Comment:
I've added support to resolve unresolved links via `documentLink/resolve` now, tested and confirm that it works with LSP-Dockerfile :slightly_smiling_face:

There are no problems when called via the command from the menu, but the hover popup will only work after I hover a second time over the link. Could someone help me and explain how I could run the request from the highlighted lines in a blocking way? I've tried with `Promise.then()` as shown here, which still seems to run async, and I've tried `Promise.resolve()` which doesn't work at all.

Reply:
There are two ways to handle it:
 - wait for all requests to finish and then show a popup
 - show the popup as soon as possible and then update it when document link + resolve request finishes

I suppose we want the latter as this functionality is not essential enough to delay the popup?
The downside of that is that the content will visibly change (potentially shift) after showing but that's not a big deal IMO and if the link element is the one at the very bottom then it's a complete non-issue.

Reply:
Thanks, it works great!